class_13_parallel_programming/homework/homework.py

```diff
@@ -0,0 +1,291 @@
+import argparse
+import io
+import logging
+import os
+import sys
+import time
+from contextlib import contextmanager
+from functools import wraps
+from multiprocessing import Lock
+from multiprocessing.pool import ThreadPool
+from pathlib import Path, PurePath
+from typing import Any, Iterable, List, Tuple
+
+import requests
+from PIL import Image
+
+# not sure how to set it up correctly. Used by `handle_image_processing` function
+ProcessParams = Tuple[str, Tuple[int, int], int, Lock, Iterable[Tuple[int, int]]]
```

Comment:
Just use namedtuple from collections module
You can setup it like this:
`ProcessParams = namedtuple('ProcessParams', ['url', 'size', 'n', 'lock', 'glob_count'])`
Create like this:
`ProcessParams(url, args.size, n, lock, counter)`
And use like this:
`resize_save_img(img_from_url, params.size, params.n)`